src/initialize.cpp

```diff
@@ -0,0 +1,226 @@
+#include "initialize.h"
+
+#include <cstddef>
+#include <cstring>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "error.h"
+#include "hdf5_interface.h"
+#include "message_passing.h"
+#include "openmc.h"
+#ifdef _OPENMP
+#include "omp.h"
+#endif
+
+// data/functions from Fortran side
+extern "C" bool openmc_check_overlaps;
+extern "C" bool openmc_write_all_tracks;
+extern "C" bool openmc_particle_restart_run;
+extern "C" bool openmc_restart_run;
+extern "C" void print_usage();
+extern "C" void print_version();
+
+
+// Paths to various files
+extern "C" {
+  char* openmc_path_input;
+  char* openmc_path_statepoint;
+  char* openmc_path_sourcepoint;
+  char* openmc_path_particle_restart;
+  bool is_null(void* ptr) {return !ptr;}
+}
+
+int openmc_init(int argc, char* argv[], const void* intracomm)
```

Comment:
This function is called by the PyAPI CAPI, right?  If so, stick an `extern "C"` on there.

Reply:
`openmc_init` is declared in `include/openmc.h` where it is in an `extern "C"` block. At the point of definition, it's not necessary to include the `extern "C"`.